firebase-firestore/src/main/java/com/google/firebase/firestore/local/TargetData.java

```diff
@@ -158,7 +187,8 @@ public boolean equals(Object o) {
         && purpose.equals(targetData.purpose)
         && snapshotVersion.equals(targetData.snapshotVersion)
         && lastLimboFreeSnapshotVersion.equals(targetData.lastLimboFreeSnapshotVersion)
-        && resumeToken.equals(targetData.resumeToken);
+        && resumeToken.equals(targetData.resumeToken)
+        && expectedCount == targetData.expectedCount;
```

Comment:
This `==` comparison here is not correct. If both `expectedCount` and `targetData.expectedCount` are distinct `Integer` objects but have the same underlying integer value then they will compare unequal using the `==` operator. We want objects to compare equal in this case.

Instead, use `Objects.equals(...)` to compare the objects for equality. It will correctly handle null values and use `.equals()` if they are both non-null.

https://developer.android.com/reference/java/util/Objects?hl=en#equals(java.lang.Object,%20java.lang.Object)